### PR TITLE
[stdlib] NFC: Remove non-Apple dead code

### PR DESCRIPTION
**Pull request by davezarzycki**
This code is dead due to: cbf157f9247fe59021c550f20f9593ed5b277d6e (#14921)

### CODE REVIEW

File: stdlib/public/stubs/UnicodeNormalization.cpp [hunk 1]
```diff
@@ -78,31 +78,6 @@ UBool u_isdefined(UChar32);
...
-static const UCollator *MakeRootCollator() {
-  UErrorCode ErrorCode = U_ZERO_ERROR;
-  UCollator *root = ucol_open("", &ErrorCode);
-  if (U_FAILURE(ErrorCode)) {
-    swift::crash("ucol_open: Failure setting up default collation.");
-  }
-  ucol_setAttribute(root, UCOL_NORMALIZATION_MODE, UCOL_ON, &ErrorCode);
-  ucol_setAttribute(root, UCOL_STRENGTH, UCOL_TERTIARY, &ErrorCode);
-  ucol_setAttribute(root, UCOL_NUMERIC_COLLATION, UCOL_OFF, &ErrorCode);
-  ucol_setAttribute(root, UCOL_CASE_LEVEL, UCOL_OFF, &ErrorCode);
```
Comment (lorentey) [inline stdlib/public/stubs/UnicodeNormalization.cpp]: I suspect that removing these may allow us to also remove some of the `#include`s above; perhaps `unicode/ucol.h` and/or `unicode/ucoleitr.h`?

**Review by lorentey — APPROVED** [review]
Nice catch!

**Review by lancep — APPROVED** [review]
Thanks for noticing this!

**Comment by lancep** [comment]
@swift-ci please test

File: stdlib/public/stubs/UnicodeNormalization.cpp (hunk 1 above)
Comment (davezarzycki) [reply stdlib/public/stubs/UnicodeNormalization.cpp]: I haven't tried. I just silenced the warning.

**Review by milseman — APPROVED** [review]
LGTM

**Comment by lancep** [comment]
@swift-ci please test Linux platform

**Comment by davezarzycki** [comment]
The Linux failure looks like unrelated LLDB problems. I'm not sure what retesting the platform will accomplish. What am I missing?

**Comment by lancep** [comment]
It appears to be a random, unrelated failure. For another PR I had, retesting got past it

**Comment by davezarzycki** [comment]
@swift-ci please smoke test linux

**Comment by lorentey** [comment]
@swift-ci please test

**Pull request merged**